ITR/data/base_providers.py

```diff
@@ -717,121 +821,91 @@ def project_ei_targets(self, company: ICompanyData, production_bm: pd.Series) ->
-                                      f"{company.company_name} is estimated with trajectory projection.")
-
-                    last_year, value_last_year = last_year_data.year, last_year_data.value
+                        last_ei_year = target.target_base_year
+                        last_ei_value = Q_(target.target_base_year_qty, target.target_base_year_unit)
+
                     target_year = target.target_end_year
                     # Attribute target_reduction_pct of ITargetData is currently a fraction, not a percentage.
-                    target_value = Q_(target.target_base_year_qty * (1 - target.target_reduction_pct),
-                                      target.target_base_year_unit)
-                    CAGR = self._compute_CAGR(value_last_year, target_value, (target_year - last_year))
-                    ei_projections = [ICompanyEIProjection(year=year, value=value_last_year * (1 + CAGR) ** (y + 1))
-                                      for y, year in enumerate(range(1 + last_year, 1 + target_year))]
-                    if ei_projection_scopes[scope] is not None:
-                        ei_projection_scopes[scope].projections.extend(ei_projections)
-                    else:
-                        ei_projection_scopes[scope] = ICompanyEIProjections(projections=ei_projections,
-                                                                            ei_metric=IntensityMetric.parse_obj({'units': target.target_base_year_unit}))
+                    target_ei_value = Q_(target.target_base_year_qty * (1 - target.target_reduction_pct),
+                                         target.target_base_year_unit)
+                    CAGR = self._compute_CAGR(last_ei_value, target_ei_value, (target_year - last_ei_year))
+                    model_ei_projections = [ICompanyEIProjection(year=year, value=last_ei_value * (1 + CAGR) ** (y + 1))
+                                            for y, year in enumerate(range(last_ei_year, 1+target_year))
+                                            if year >= self.projection_controls.BASE_YEAR]
                 elif target.target_type == "absolute":
                     # Complicated case, the target must be switched from absolute value to intensity.
                     # We use the benchmark production data
                     # Compute Emission CAGR
-                    emissions_data = historic_data.emissions.__getattribute__(scope)
-
-                    # Get last year data with non-null value
-                    if ei_projection_scopes[scope] is not None:
-                        last_year_ei = ei_projection_scopes[scope].projections[-1]
-                        last_year = last_year_ei.year
-                        last_year_prod = production_bm.loc[last_year]
-                        last_year_data = IEmissionRealization(year=last_year, value=last_year_ei.value*last_year_prod)
```

Comment:
you removed call to `IEmissionRealization` - does it still work?
was this call unnecessary in the original code?

Reply:
EEmissionsRealization is just a constructor that creates a BaseModel-ish version of a tuple.  It's a real pain to work with compared with real tuples.  So I rewrote the code to avoid it, using pythonic tuples instead.  You can see that the call was also removed from the following code, which also slightly simplifies things.